docs/designs proposals/authentication service/authentication_service.md

```diff
@@ -0,0 +1,13 @@
+# Authentication service
+
+## Overview
+
+Use a token based system for the authentication of devices. The authentication service can support multiples authentication mechanisms and hands out a token if authentication is sucessfull. 
+In order to avoid the authentication process for each message, the token have a limited *validity period* and are cached by the adapters. 
```

Comment:
```suggestion
In order to avoid the authentication process for each message, the token have a limited *validity period* and are cached by the devices. 
```

I guess this is just a typo. The drawing already forwards the token to the device, which would need to keep/cache the token for later use.

Reply:
the idea i had was that the adapter could have a cache of the token signature in order to validate the signature for subsequent times. This would remove the auth service from the critical path. Obviously the cache entries would have an expiration timer

Reply:
> the idea i had was that the adapter could have a cache of the token signature in order to validate the signature for subsequent times.

I am not sure I understand. So when the device connects to the adapter, what would happen?

Reply:
it would send the Json Web Token along with the request. The adapter lookup the signature part and compares that it's the same as the original signature, if so, forward the data to kafka. If the cached entry (with the signature) expires, the device must renew the token with the device registry